src/core/Akka/IO/TcpConnection.cs

```diff
@@ -815,73 +827,169 @@ private sealed class PendingBufferWrite : PendingWrite
                 IActorRef self,
                 IActorRef commander,
                 ByteString buffer,
-                object ack,
-                WriteCommand tail) : base(commander, ack)
+                object ack) : base(commander, ack)
             {
                 _connection = connection;
                 _sendArgs = sendArgs;
                 _self = self;
                 _buffer = buffer;
-                _tail = tail;
 
                 // start immediatelly as we'll need to cover the case if 
                 // after buffer write request, the remaining enumerator is empty
                 //hasData = this.remainingData.MoveNext();
             }
 
-            public override PendingWrite DoWrite(ConnectionInfo info)
+            public override void DoWrite(ConnectionInfo info)
             {
                 try
                 {
-                    var data = _buffer;
-                    while(true)
-                    {
-                        var bytesWritten = Send(data);
-
-                        if (_connection.traceLogging)
-                            _connection.Log.Debug("Wrote [{0}] bytes to channel", bytesWritten);
-                        if (bytesWritten < data.Count)
-                        {
-                            // we weren't able to write all bytes from the buffer, so we need to try again later
-                            data = data.Slice(bytesWritten);
-                        }
-                        else // finished writing
-                        {
-                            if(Ack != NoAck.Instance) Commander.Tell(Ack);
-                            Release();
-                            return _connection.CreatePendingWrite(Commander, _tail, info);
-                        }
-                    }
-
+                    _sendArgs.SetBuffer(_buffer);
+                    if (!_connection.Socket.SendAsync(_sendArgs))
+                        _self.Tell(SocketSent.Instance);
+                    
+                    Release();
                 }
                 catch (SocketException e)
                 {
                     _connection.HandleError(info.Handler, e);
-                    return this;
                 }
             }
 
-            private int Send(ByteString data)
+            public override void Release() { }
```

Comment:
Because if that's the case, we need to clear away the old content here and make this object re-usable again.

Reply:
See my reply above.